go-controller/pkg/metrics/ovnkube_controller.go

```diff
@@ -649,6 +671,49 @@ func ipsecMetricHandler(table string, model model.Model) {
 	}
 }
 
+// MontiorIPsecTunnelsState will register a labeled metric for the ipsec tunnel status. It will keep
+// checking ipsec established status for each Geneve tunnel towards other remote node for every 5s
+// and update the metric with tunnel up or down status.
+func MontiorIPsecTunnelsState(stopChan <-chan struct{}, nbClient libovsdbclient.Client) {
+	prometheus.MustRegister(metricIPsecTunnelState)
+	go func() {
+		for {
+			select {
+			case <-stopChan:
+				return
+			default:
+				nbGlobal := &nbdb.NBGlobal{}
+				nbGlobal, err := libovsdbops.GetNBGlobal(nbClient, nbGlobal)
+				if err != nil {
+					klog.Errorf("Error in getting the NBGlobal row from Northbound db, err: %v", err)
+					return
+				}
+				if !nbGlobal.Ipsec {
+					// IPsec is not enabled, so keep checking for ipsec after wait time.
+					time.Sleep(ipSecTunnelStatusQueryWaitTime)
```

Comment:
This a go routine. Youre blocking it here and it cannot exit. Also, theres no way for this go routine to say I'm done, you can exit, we usually use wait groups. Is it possible ?

Why not expose its true value directly instead of waiting around for it to be true ?

Reply:
oh! yes, now gracefully exiting using wait group.

we don't need to report this metric if ipsec is not enabled by user, hence this check is needed.